lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -545,42 +533,28 @@ export default class BitcoinProcessor {
 
     console.info(`Reverting quantile and transaction DBs to closest fee sampling group boundary given block: ${lastValidBlockNumber}`);
 
-    // For the quantile DB, we need to remove the full group (corresponding to the given
-    // lastValidBlockNumber). This is because currently there is no way to add/remove individual block's
-    // data to the quantile DB ... it only expects to work with the full groups. Which means that we
-    // need to remove all the transactions which belong to that group (and later ones).
-    //
-    // So what we need to do is:
-    // <code>
-    //   validBlockGroup = findGroupForTheBlock(lastValidBlockNumber);
-    //   firstBlockInGroup = findFirstBlockInGroup(validBlockGroup);
-    //
-    //   deleteAllTransactionsGreaterThanOrEqualTo(firstBlockInGroup);
-    //   deleteAllGroupsGreaterThanOrEqualTo(validBlockGroup);
-    // </code>
-    const firstBlockInGroup = this.getFirstBlockInGroup(lastValidBlockNumber);
+    // Basically, we need to remove all the transactions/normazlied-fee-data from the system later than the
+    // specified lastValidBlockNumber input.
+
+    // Get the first block and transaction from the group which are supposed to be deleted
+    const firstTxnOfGroup = this.normalizedFeeCalculator.getFirstTransactionOfGroup(lastValidBlockNumber);
+    const firstBlockInGroup = TransactionNumber.getBlockNumber(firstTxnOfGroup);
 
     // NOTE:
     // *****
-    // Make sure that we remove the transaction data BEFORE we remove the quantile data. This is
+    // Make sure that we remove the transaction data BEFORE we remove the normalized data. This is
     // because that if the service stops at any moment after this, the initialize code looks at
     // the transaction store and can revert the quantile db accordingly.
-    const firstTxnOfFirstBlockInGroup = TransactionNumber.construct(firstBlockInGroup, 0);
-
-    console.debug(`Removing transactions since ${firstBlockInGroup} (transaction id: ${firstTxnOfFirstBlockInGroup})`);
-    await this.transactionStore.removeTransactionsLaterThan(firstTxnOfFirstBlockInGroup - 1);
 
-    // Now revert the corresponding groups (and later) from the quantile calculator.
-    const revertToGroupId = this.getGroupIdFromBlock(firstBlockInGroup);
+    // Remove all the txns which are in that first block (and greater)
+    console.debug(`Removing transactions since ${firstBlockInGroup} (transaction id: ${firstTxnOfGroup})`);
```

Comment:
Nit: since seems to include exclusivity.
```suggestion
    console.debug(`Removing transactions since and including ${firstBlockInGroup} (transaction id: ${firstTxnOfGroup})`);
```